src/helpers/strings.ts

```diff
@@ -130,3 +179,112 @@ export function formatCurrency(
 
   return `${currencySymbol}${formattedWhole}.${formattedFraction}`;
 }
+
+export function formatFractionWorklet(fraction: string): string {
+  'worklet';
+  let leadingZeros = 0;
+  for (let i = 0; i < fraction.length; i++) {
+    if (fraction[i] === '0') {
+      leadingZeros++;
+    } else {
+      break;
+    }
+  }
+
+  const significantPart = fraction.slice(leadingZeros);
+  if (significantPart.length === 0 || /^[0]+$/.test(significantPart)) {
+    return '00';
+  }
+
+  const significantDigits = significantPart.slice(0, 2).padEnd(2, '0');
+
+  if (leadingZeros >= 4) {
+    const leadingZerosStr = leadingZeros.toString();
+    let subscriptZeros = '';
+    for (let i = 0; i < leadingZerosStr.length; i++) {
+      subscriptZeros += toSubscript(String(leadingZerosStr[i]));
+    }
+    return `0${subscriptZeros}${significantDigits}`;
+  }
+  return `${'0'.repeat(leadingZeros)}${significantDigits}`;
+}
+
+const SUBSCRIPT_THRESHOLD_MAGNITUDE = -4;
+
+export function toCompactNotation({
+  value,
+  prefix,
+  decimalPlaces,
+}: {
+  value: string | number;
+  prefix?: string;
+  decimalPlaces?: number;
+}): string {
```

Comment:
This needs to handle the currency formatting for thousands separators either by default or as an option, no longer seeing comma on eth chart. 

<img width="257" alt="Screenshot 2025-04-11 at 3 43 01 PM" src="https://github.com/user-attachments/assets/85dc7989-432b-4cf4-95c2-5de8c8079d2c" />